include/aikido/planner/vectorfield/VectorFieldConfigurationToEndEffectorOffsetPlanner.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_VECTORFIELDCONFIGURATIONTOENDEFFECTOROFFSETPLANNER_HPP_
+#define AIKIDO_PLANNER_VECTORFIELD_VECTORFIELDCONFIGURATIONTOENDEFFECTOROFFSETPLANNER_HPP_
+
+#include "aikido/planner/ConfigurationToEndEffectorOffset.hpp"
+#include "aikido/planner/ConfigurationToEndEffectorOffsetPlanner.hpp"
+#include "aikido/planner/vectorfield/VectorFieldPlanner.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planner that generates a trajectory that moves the end-effector by a given
+/// direction and distance.
+class VectorFieldConfigurationToEndEffectorOffsetPlanner
+    : public ConfigurationToEndEffectorOffsetPlanner
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] stateSpace State space that this planner is associated with.
+  /// \param[in] metaskeleton MetaSkeleton to plan with.
+  /// \param[in] distanceTolerance How much a planned trajectory is allowed to
+  /// deviate from the requested distance to move the end-effector.
+  /// \param[in] positionTolerance How a planned trajectory is allowed to
+  /// deviated from a straight line segment defined by the direction and the
+  /// distance.
+  /// \param[in] angularTolerance How a planned trajectory is allowed to deviate
+  /// from a given direction.
+  /// \param[in] initialStepSize Initial step size.
+  /// \param[in] jointLimitTolerance If less then this distance to joint
+  /// limit, velocity is bounded in that direction to 0.
+  /// \param[in] constraintCheckResolution Resolution used in constraint
+  /// checking.
+  /// \param[in] timelimit timeout in seconds.
+  VectorFieldConfigurationToEndEffectorOffsetPlanner(
+      statespace::ConstStateSpacePtr stateSpace,
+      dart::dynamics::MetaSkeletonPtr metaskeleton,
+      double distanceTolerance,
+      double positionTolerance,
+      double angularTolerance,
+      double initialStepSize,
+      double jointLimitTolerance,
+      double constraintCheckResolution,
+      std::chrono::duration<double> timelimit);
```

Comment:
Parameters that are not planner specific parameters should move to the problem class. @dqyi11 Could you verify all the parameters are planner specific?

Reply:
I agree with the current scope of parameters defined in the problem class, which is consistent  with the design in `prpy`.  https://github.com/personalrobotics/prpy/blob/master/tests/planning/methods/PlanToEndEffectorOffset.py
All these parameters are planner-specific. Though some of them are used in both `vector field planner` and `CRRT planner`, they can be set very differently (for example `constraintCheckResolution`). `distanceTolerance`, `positionTolerance` and  `angularTolerance` are also not strictly aligned in planning end-effector offset between using vector field planner and CRRT planner.